### DISCUSSION TITLE
Methods to create & data on hourly demand time series

### DISCUSSION THREAD

**Discussion by pz-max** [pypsa-meets-earth/pypsa-earth#63] [General]
We need ideally hourly demand time series that are aggregated to rasters (i.e. 25x25km).

Some general methods we could observe:
- [GEGIS](https://github.com/niclasmattsson/GlobalEnergyGIS#synthetic-electricity-demand) & its synthetic demand time series estimates.
- [Top-down approach](https://www.worldpop.org/methods/populations). Disaggregate national aggregated hourly demand time series based on GDP/population raster similar as applied in [PyPSA-Eur](https://github.com/PyPSA/pypsa-eur#pypsa-eur-an-open-optimisation-model-of-the-european-transmission-system).
- [Bottom-up approach](https://www.worldpop.org/methods/populations). Use building wise demand estimations and scale them up to the raster. [RAMP might help here](https://github.com/RAMP-project/RAMP) in combination with OpenStreetMap building data. The OSM data coverage seems ok:

![image](https://user-images.githubusercontent.com/61968949/128027184-99abcd50-049c-4271-93d9-2e12f20ca1a0.png)

**Comment by euronion** [comment]
GEGIS is an easy solution for a prototype. I've worked with it before, so it was easy for me to create a more universal workflow, see the dedicated repository here: https://github.com/euronion/synde/
The amount of data to download and computational intensity is rather high, so it takes a day or two to create the first demand series.

Therefore I'll create a few batches (for different scenarios) and upload them somewhere for direct usage.

Advantage of GEGIS is, that it makes use of the SSP scenarios for determining GDP/population and correlated electricity demand.
Disadvantage: Only electricity demand and generalised approach (not country specific forecast methodology).

**Comment by pz-max** [comment]
Just an alternative to GEGIS if someone wants to spend some time on demand modelling. Giacomo Falchetta has developed and applied tools to better understand electricity access and estimate electricity demand in the region at high spatio-temporal resolution using GIS and satellite-based information.

See some recent publications which might help:
https://iopscience.iop.org/article/10.1088/1748-9326/ac0cab
https://www.nature.com/articles/s41597-019-0122-6
https://www.sciencedirect.com/science/article/pii/S0140988321002127

**Reply by pz-max** [reply]
@Nayara2020

**Reply by pz-max** [reply]
Thierry

**Comment by pz-max** [comment]
The story for why demand prediction is needed:
- Energy system models are used for GHG predictions and net-zero transition pathways.
- These models are impacted by the future climate.
- Further, energy demand is a huge uncertainty factor in Africa models and also climate-related
- We use AI to predict the climate and energy demand.
- The predictions are used in energy system models and impact policymaker decisions

## Electricity demand prediction proposal using ML
Generally, the more data, the better. To run the ML methods successfully, data worldwide must be collected and leveraged. Requesting some more data from Africa and other countries can help the accuracy as previously mentioned. The goal is to predict better hourly and yearly demand profiles.

Here are relevant papers which also contain supplemental material with data:
 - Toktarova et al. (2019) - closed-source but shared some data: https://www.sciencedirect.com/science/article/pii/S0142061518336196#!
 - Mattson et al. (2021) - open-source:  https://www.sciencedirect.com/science/article/pii/S2211467X20301590
 - Of potential interest: A package using ANNs to model electricity demand in the US. Pretty reliant on external data though. https://github.com/openjournals/joss-reviews/issues/4472

Training data input that is available for a given area X. X is provided as geojson (or .nc):
- hourly electricity demand profiles (often only provided on a national level, however, training data on regional/city level could help as well)
- hourly temperature profiles (27 x 27 km2)
- hourly wind speed profiles (27 x 27 km2)
- humidity (27 x 27 km2)
- population data (1 x 1 km2)
- GDP (1 x 1 km2)
- building density or coverage from OpenStreetMap (km2 coverage is given area)-
